### PR TITLE
refactor: do not generate JSON with string templates

### LINKED ISSUE #28: Refactor generators to use a JSON builder
The generators should be implemented with a builder instead of string templates. That way the methods will cleaner, not duplicated as much and reusable.

### PR DESCRIPTION
**Pull request by tpluscode**
closes #28

### CODE REVIEW

**Comment by tpluscode** [comment]
More code removed than added. That is always good, right? :)

**Comment by tpluscode** [comment]
In fact not Java. The language is Xtend

File: app.hypermedia.testing.dsl/src/main/java/app/hypermedia/testing/dsl/generator/CoreGenerator.xtend [hunk 1]
```diff
@@ -34,116 +40,112 @@ class CoreGenerator extends AbstractGenerator {
...
+        	steps.put(new JSONObject(stepJson.toString()))
         }
-    '''
+        
+        scenario.put('steps', steps)
+        return scenario
+    }
+    
+    def buildBlock(String type, Iterable<? extends EObject> children, Map<String, Object> map) {   
+        map.put('children', children.map([c | c.step as JSONObject]).toList())
```
Comment (mchlrch) [inline app.hypermedia.testing.dsl/src/main/java/app/hypermedia/testing/dsl/generator/CoreGenerator.xtend]: is casting `c.step as JSONObject` necessary?

Comment (tpluscode) [reply app.hypermedia.testing.dsl/src/main/java/app/hypermedia/testing/dsl/generator/CoreGenerator.xtend]: In fact it isn't. Xtend complained when `map` was called in a different place, but not here actually :+1:.

**Pull request merged**